parsl/executors/high_throughput/mpi_prefix_composer.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from typing import Dict, List, Tuple, Set
+
+logger = logging.getLogger(__name__)
+
+VALID_LAUNCHERS = ('srun',
+                   'aprun',
+                   'mpiexec')
+
+
+class InvalidResourceSpecification(Exception):
+    """Exception raised when Invalid keys are supplied via resource specification"""
+
+    def __init__(self, invalid_keys: Set[str]):
+        self.invalid_keys = invalid_keys
+
+    def __repr__(self):
+        return f"Invalid resource specification options supplied: {self.invalid_keys}"
```

Comment:
repr shouldn't be a human-targeted string: see #1966 and https://docs.python.org/3/library/functions.html#repr

The default `repr` should be ok for exceptions and this human readable text can go into `__str__`.